src/backend/distributed/commands/multi_copy.c

```diff
@@ -3547,7 +3553,21 @@ CopyGetPlacementConnection(HTAB *connectionStateHash, ShardPlacement *placement,
 	if (placement->partitionMethod == DISTRIBUTE_BY_HASH &&
 		MultiShardConnectionType != SEQUENTIAL_CONNECTION)
 	{
-		connectionFlags |= CONNECTION_PER_PLACEMENT;
+		/*
+		 * Claiming the connection exclusively (done below) would also have the
+		 * effect of opening multiple connections, but claiming the connection
+		 * exclusively prevents GetConnectionIfPlacementAccessedInXact from returning
+		 * the connection if it is needed for a different shard placement.
+		 *
+		 * By setting the REQUIRE_CLEAN_CONNECTION flag we are guaranteed to get
+		 * connection that will not be returned by GetConnectionIfPlacementAccessedInXact
+		 * for the remainder of the COPY, hence it safe to claim the connection
+		 * exclusively. Claiming a connection exclusively prevents it from being
+		 * used in other distributed queries that happen during the COPY (e.g. if
+		 * the copy logic calls a function to calculate a default value, and the
+		 * function does a distributed query).
+		 */
+		connectionFlags |= REQUIRE_CLEAN_CONNECTION;
```

Comment:
I suggest writing a method for setting this flag and putting the part of the comment that is purely about what this flag is doing. So that we don't have to write the same comment when we are setting this flag, we can just call the function. And here the comment can explain only why we are calling that function.

Reply:
Thus flag is only used here, and not likely to be used anywhere else. 

Finding a good name  for such a function is hard and puts additional complexity. I'd suggest we do it if we ever  re-use this flag.

We recently discussed this with @pykello, if it is not east to find a good name for a function, then it is probably much better not to move it into a function -- especially  for these type of few lines of code blocks.